File: Lists/ReverseNode/ReverseNode.scala
/**
 * Definition for singly-linked list.
 * class ListNode(_x: Int = 0, _next: ListNode = null) {
 *   var next: ListNode = _next
 *   var x: Int = _x
 * }
 */
object Solution {
    def swapPairs(head: ListNode): ListNode = {
        if (head == null || head.next == null) return head
        else {
        val firstNode = head
        val secondNode = head.next
        
        firstNode.next  = swapPairs(secondNode.next);
        secondNode.next = firstNode;
        
        return secondNode}
    }
}

File: Recursion/ReverseNode/ReverseNode.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if not head or not head.next:
            return head
        
        firstNode = head
        secondNode = head.next
        
        firstNode.next = self.swapPairs(secondNode.next)
        secondNode.next = firstNode
        
        return secondNode